src/sentry/static/sentry/app/views/alerts/list/index.tsx

```diff
@@ -99,7 +152,7 @@ class IncidentsList extends AsyncComponent<Props, State & AsyncComponent['state'
           </PanelHeader>
 
           <PanelBody>
-            {loading && <LoadingIndicator />}
+            {(loading || this.state.hasAlertRule === undefined) && <LoadingIndicator />}
```

Comment:
I don't think this is quite right, the "loading" state should be a combination of 1) `this.state.loading`, 2) not loading and no incidents and `hasAlertRule` is undefined.

Reply:
Right, good catch. I was forcing us to wait for the rules check even if we had incidents.